benchmarks/erp12/cbgp_lite/benchmark/suite/composite.clj

```diff
@@ -0,0 +1,523 @@
+                                 upper (lib/max' x y)]
+                             {:inputs [the-set lower upper]
+                              :output (set (filter #(and (lib/<' lower %) (lib/<' % upper))
+                                                   the-set))})))
+       :loss-fns       [bu/jaccard-similarity-loss]}
+
+      "area-of-rectangle"
+      {:description    "Given two tuples of floats representing the upper-right and
+                     lower-left coordinates of a rectangle in the cartesian plane,
+                     find the area of the rectangle."
+       :input->type    {'input1 {:type :tuple, :children [{:type 'double?} {:type 'double?}]}
+                        'input2 {:type :tuple, :children [{:type 'double?} {:type 'double?}]}}
+       :ret-type       {:type 'double?}
+       :other-types    []
+       :extra-genes    []
+       :case-generator (fn area-of-rectangle-gen
+                         []
+                         (let [xs [(rand-float-range -100 100) (rand-float-range -100 100)]
+                               ys [(rand-float-range -100 100) (rand-float-range -100 100)]
+                               x1 (reduce max xs)
+                               x2 (reduce min xs)
+                               y1 (reduce max ys)
+                               y2 (reduce min ys)
+                               output (* (- x1 x2)
+                                         (- y1 y2))]
+                           {:inputs [[x1 y1] [x2 y2]]
+                            :output output}))
+       :loss-fns       [#(bu/round 4 (bu/absolute-distance %1 %2))]}
+
+      "sum-vector-vals"
+      {:description    "Given a map {string => int} and vector of strings that are
+                         keys of the map, look up the values associated with those
+                         keys in the map and return their sum."
+       :input->type    {'input1 {:type :map-of, :key {:type 'string?}, :value {:type 'int?}}
+                        'input2  {:type :vector :child {:type 'string?}}}
+       :ret-type       {:type 'int?}
+       :other-types    [{:type 'string?} {:type 'boolean?}]
+       :extra-genes    [{:gene :lit, :val 0, :type {:type 'int?}}]
+       :case-generator (fn sum-vector-vals-gen []
+                         (let [the-map (first (:inputs (sum-2-vals-case-generator (bu/string-generator 10))))
+                               prob (+ 0.1 (rand 0.8))
+                               the-vector (vec (random-sample prob (keys the-map)))]
+                           {:inputs [the-map the-vector]
+                            :output (apply + (map the-map the-vector))}))
+       :loss-fns       [bu/absolute-distance]}
+
+      "sets-with-element"
+      {:description    "Given a set of sets, filter to only contain sets that
+                         contain a certain element."
+       :input->type    {'input1 {:type :set :child {:type :set :child {:type 'int?}}}
+                        'input2  {:type 'int?}}
```

Comment:
It could be, but I don't think it needs to be, and it would change the problem -- it would create a smaller instruction set, since int-specific instructions wouldn't be present, and you wouldn't include an int ERC. It would also require changing the random set generator, which sounds like a pain. I'd vote keep it as-is.

Reply:
Sounds good. Let's leave it.